### PR TITLE
Add full userguide

### PR DESCRIPTION
**Pull request by cryptointerest**
Add the new user guide.

### CODE REVIEW

**Comment by ShadowMitia** [comment]
What is the status on this?

**Comment by cryptointerest** [comment]
the status is that I wrote this user guide and while doing this succesively changed the code base but was not entirely clean with my commits. 
I also introduced a makefile for which Stephan requested changes. 

The problem is, I still have tendinitis issues in both arms and absolutely cannot (or rather must not) code in my free time. 
It bugs me because the work is done, but I can't work on cleaning up my commit history atm. 
I'll try to see if I can do it over Christmas

**Comment by ShadowMitia** [comment]
Take time to rest and recuperate!
Let us know what we can do to help.

If it's just rebase/squashing commits maybe I can do it for you? If it's not too complicated to explain for you.
Otherwise there is no hurry, it just seemed like it was all done and could possibly be merged.

**Comment by cryptointerest** [comment]
I'll have to have a look. If I remember correctly, I was not clean with my commits in the sense that i staged changes to the code AND the guide (Thought it was a good idea at the time. Still learning, I guess).
So to clean up things some work needs to be done.

File: manual/moderncv_userguide.tex [hunk 1]
```diff
@@ -1,462 +1,996 @@
...
+\begin{lstlisting}
+  \cvlistitem{<Item>}
+\end{lstlisting} 
+
+\paragraph{\code{\\cvlistdoubleitem.}} Command taking two argument intended typeset a bullet list in two columns. Usage
+\begin{lstlisting}
+  \cvlistdoubleitem{<Item_1>}{Item_2>}
+\end{lstlisting} 
+
+\paragraph{\code{cvcolumns.}} Environment to tipeset multi-column cvitems. This can be combined with the itemize environment.
```
Comment (nsbruce) [inline manual/moderncv_userguide.tex:473]: typo - should be typeset

File: manual/moderncv_userguide.tex [hunk 2]
```diff
@@ -1,462 +1,996 @@
...
+This will leave the definitions of \Code{itemize} environments untouched.
+\end{itemize}
 
-\section{More customizations}
+\subsection{Adjusting lengths}
+\todo{Add more adjustable lengths}
+Some of the lengths in moderncv can be adjusted. 
+The hints column can be adjusted by issuing
+\begin{lstlisting}
+  \setlength{\hintscolumnwidth}{3cm}
```
Comment (nsbruce) [inline manual/moderncv_userguide.tex:766]: showing the default for each length in this section (as you do for `\cvsectionstretchability`) would be beneficial

**Comment by stephanlachnit** [comment]
Closing in favor of #106.

**Pull request closed without merging**